modules/launcher/src/main/java/org/mule/runtime/module/launcher/log4j2/MuleLog4jContextFactory.java

```diff
@@ -53,12 +56,22 @@
   private static final String ASYNC_LOGGER_EXCEPTION_HANDLER_PROPERTY = "AsyncLoggerConfig.ExceptionHandler";
   private static final String DEFAULT_ASYNC_LOGGER_EXCEPTION_HANDLER = AsyncLoggerExceptionHandler.class.getName();
 
+  private static final boolean LOG_SEPARATION_ENABLED = getProperty(MULE_LOG_SEPARATION_DISABLED) == null;
```

Comment:
The exact same condition is already evaluated in `MuleContainer` class. Extract.